examples/todoapp/lib/base/di/todoapp_with_dependencies.dart

```diff
@@ -107,22 +109,28 @@ class TodoappWithDependencies extends StatelessWidget {
 
   List<SingleChildWidget> get _libs => [
         ...TranslationsDependencies.from(baseUrl: config.baseUrl).providers,
+        ...RealmDependencies().providers,
       ];
 
   List<Provider> get _dataSources => [
+        Provider<ConnectivityDataSource>(
+          create: (context) => ConnectivityDataSource(),
+        ),
+        Provider<TodoLocalDataSource>(
+          create: (context) => TodoLocalDataSource(
+            context.read<RealmService>().realm,
+          ),
+        ),
         Provider<TodosRemoteDataSource>(
           create: (context) => TodosRemoteDataSource(
             context.read<ApiHttpClient>(),
           ),
         ),
         Provider<PermissionsRemoteDataSource>(
-          create: (context) => PermissionsRemoteDataSource(
-            context.read<ApiHttpClient>(),
-          ),
+          create: (context) => PermissionsRemoteDataSource(),
```

Comment:
Why the permission data source does not depend on the http client ?

Reply:
Fixed